projects/storefrontlib/src/layout/a11y/skip-links/directive/skip.directive.ts

```diff
@@ -0,0 +1,29 @@
+import {
+  Directive,
+  TemplateRef,
+  Input,
+  OnDestroy,
+  AfterViewInit,
+} from '@angular/core';
+import { SkipLinkService } from '../skip-link.service';
+
+@Directive({
+  selector: '[cxSkip]',
+})
+export class SkipDirective implements AfterViewInit, OnDestroy {
+  @Input()
+  cxSkip: string;
+
+  constructor(
+    private tpl: TemplateRef<any>,
```

Comment:
What do you mean by bringing in the elementRef?

Reply:
I was worried about the fact that you currently use the directive on an `ng-template` only. Perhaps you can try to use it locally on an ordinary `div` element, and see if you're able to make it work. Most likely you need to use the `elementRef` instead of the `TemplateRef`, but I might be wrong.